Jenkinsfile-deploy

```diff
@@ -0,0 +1,59 @@
+pipeline {
+  agent none
+  parameters {
+    choice(name: "STAGE", choices: ["intg", "staging", "prod"], description: "The stage you are building the front end for")
+  }
+  stages {
+    stage("Test") {
+        agent {
+            ecs {
+                inheritFrom "sbt"
+            }
+        }
+        steps {
+            checkout scm
+            sh "sbt dist"
+            stash includes: "Dockerfile", name: "Dockerfile"
+            stash includes: "target/universal/tdr-transfer-frontend-*.zip", name: "tdr-transfer-frontend-zip"
+        }
+    }
+    stage("Docker") {
+        agent {
+            label "master"
+        }
+        steps {
+            unstash "tdr-transfer-frontend-zip"
+            unstash "Dockerfile"
+            sh "docker build -t nationalarchives/tdr-transfer-frontend:${params.STAGE} ."
+            withCredentials([usernamePassword(credentialsId: "docker", usernameVariable: "USERNAME", passwordVariable: "PASSWORD")]) {
+                sh "echo $PASSWORD | docker login --username $USERNAME --password-stdin"
+                sh "docker push nationalarchives/tdr-transfer-frontend:${params.STAGE}"
+                slackSend color: "good", message: "The front end app has been pushed to docker hub", channel: "#tdr"
+            }
+        }
+    }
+    stage("Update ECS container") {
+        agent {
+            ecs {
+                inheritFrom "aws"
+            }            
+        }
+        steps {
+            withCredentials([[
+                $class: "AmazonWebServicesCredentialsBinding",
+                credentialsId: "integrationTerraform",
+                accessKeyVariable: "AWS_ACCESS_KEY_ID",
```

Comment:
I was talking to @TomJKing last week about how feasible it would be to use the [ECS task role](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_IAM_role.html) to avoid having to generate an AWS access key for pushing things from Jenkins. What do you think?

Reply:
We could do, this user is the user that jenkins uses to terraform the environments which obviously has permissions to update ECS services so it's not like we're creating a new one for this purpose.

Reply:
I've changed my mind, using the terraform user violates the principle of least privilege. I'll create a new user and add it to Jenkins for the ECS deployments.

Reply:
I've just realised what you mean in that we don't need a user at all, we can pass a role through to the fargate container which will then create a temporary user based on that role and use that to update the service. I've created these roles in the other terraform projects and this is now working without creating an explicit user. 
Also, this job now is not running using the terraform user which has far too many privileges, the policy against the role only allows the container to update the ECS service.

Reply:
Cheers - locking down the role used by the tasks sounds good.

Is https://github.com/nationalarchives/tdr-terraform-backend/pull/8 the change you mean?

Reply:
That's the one yeah.